crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -68,9 +68,11 @@ pub async fn check_transaction_input(
     transaction.check_version_supported(epoch_store.protocol_config())?;
     transaction.validity_check(epoch_store.protocol_config())?;
     let input_objects = transaction.input_objects()?;
+    let receiving_objects = transaction.receiving_objects()?;
```

Comment:
There are several similar functions in this file that checks input for different kinds of transactions. It's an unfortunate duplication of code but it does mean we have to handle receiving objects in all those places. This further backs my question regarding whether we should include receiving objects in input objects, otherwise you would have to chase down all calls to input_objects().

Reply:
My understanding is that receiving objects can only be used for PTBs (outside of those they don't make any sense I believe). So I don't believe we need to handle these separate apart from possibly checking that the set of `receiving_objects` is empty in that case.

Reply:
Not sure if I understand, but `check_transaction_input_with_given_gas` and `check_dev_inspect_input` could be called on PTBs as well?

Reply:
ahh correct, these will need to be added there as well.